### PR TITLE
[backend/frontend] Add new filters on inject contract panel & persist choice in local storage

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Proposed changes

* In injector contracts panel when we create a new inject
   *  Improv dynamic filters -> Add Platform, Kill Chain Phases & Injector
   * Store filter, sort & pagination in local storage
* In Scenarios lists
   *  Improv dynamic filters -> Add Kill chain phases & Tags
   * Store filter, sort & pagination in local storage
* In Simulations lists
   *  Improv dynamic filters -> Add Scenario, Kill chain phases & Tags
   * Store filter, sort & pagination in local storage

Component V2 should replace, at the end of this issue, the others components

### Related issues

* https://github.com/OpenBAS-Platform/openbas/issues/124
Chunks 1 & 2

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by savacano28** [comment]
**Todo**:  The pagination in scenarios and simulations does not show the correct total number of results.
image

**Tested**:
*In injector contracts panel when we create a new inject
    Improv dynamic filters -> Add Platform, Kill Chain Phases & Injector: OK
    Store filter, sort & pagination in local storage: OK
*In Scenarios lists
    Improv dynamic filters -> Add Kill chain phases & Tags: OK
    Store filter, sort & pagination in local storage: OK
*In Simulations lists
    Improv dynamic filters -> Add scenario, Kill chain phases & Tags: OK
    Store filter, sort & pagination in local storage: Ok

**Comment by RomuDeuxfois** [comment]
Ready for a second round.

File: openbas-model/src/main/java/io/openbas/database/model/Filters.java [hunk 1]
```diff
@@ -29,6 +33,31 @@ public static class FilterGroup {
...
+
+    public void removeByKey(@NotBlank final String filterKey) {
+      if (this.getFilters() == null) {
+        return;
+      }
+
+      List<Filter> newFilters = this.getFilters()
+          .stream()
+          .filter(filter -> !filter.getKey().equals(filterKey))
+          .toList();
```
Comment (savacano28) [inline openbas-model/src/main/java/io/openbas/database/model/Filters.java:57]: filter.getKey() or filterKey can be null?

File: openbas-model/src/main/java/io/openbas/database/model/Filters.java [hunk 2]
```diff
@@ -20,6 +22,8 @@ public enum FilterOperator {
     not_contains,
     starts_with,
     not_starts_with,
+    empty,
+    not_empty,
   }
```
Comment (savacano28) [inline openbas-model/src/main/java/io/openbas/database/model/Filters.java:27]: Add translations

File: openbas-model/src/main/java/io/openbas/database/model/Filters.java (hunk 1 above)
Comment (RomuDeuxfois) [reply openbas-model/src/main/java/io/openbas/database/model/Filters.java:57]: @Data
  public static class Filter {

    @NotNull
    private String key;
    private FilterMode mode; // Between values: name = name1 OR name = name2
    private List<String> values;
    private FilterOperator operator;
  }
  
  filter.getKey() can't be null
  
  @NotBlank final String filterKey
  filterKey can't be null

File: openbas-model/src/main/java/io/openbas/database/model/Filters.java (hunk 2 above)
Comment (RomuDeuxfois) [reply openbas-model/src/main/java/io/openbas/database/model/Filters.java:27]: Fix

**Review by savacano28 — APPROVED** [review]

**Pull request merged**